### ISSUE TITLE
Remove "Review page" from single-page forms as it is redundant

### ISSUE THREAD

**Issue by anikbrazeau** [cds-snc/platform-forms-client#4075]
## Description
Currently the review page applies to all forms even short one-page or one-question forms.

## Steps to reproduce
1. Go to GC Forms
2. Create a single page form with two questions
3. Create a multi-page form with many pages and questions
4. Test out both forms
5. See how review to check answers is redundant when there's only one page

## Expected behaviour
No need for review page when forms are very short and simple and when all answers can be checked on one page.
Make "Review" apply only to forms with multiple sections/pages.

- [ ] The current behavior is to only show the Review page if Conditional Logic is enabled. This task would add on to that, to only show the Review page if a form has groups (multiple pages).

## Links

Related #3960 #4049

**Comment by thiessenp-cds** [comment]
I thought about this for a bit and I think it makes sense. A user on a single page form can skim their filled in answers on that page.
Though for the "forms are very short and simple" I'm not sure how to measure this? But I think the task is just for multi-page forms?

The current behavior is to only show the Review page if Conditional Logic is enabled. This task would add on to that, to only show the Review page if a form has groups (multiple pages). Does that sound right?

**Comment by anikbrazeau** [comment]
Do we know what the behaviour would be for single-page existing forms that have already been published once conditional logic is released?

**Comment by anikbrazeau** [comment]
What if there was an option to "Skip Review page" ? 
That might satisfy all needs?

**Comment by Abi-Nada** [comment]
> Do we know what the behaviour would be for single-page existing forms that have already been published once conditional logic is released?

I've validated that the Review Page will not show up for existing Published Forms that we have in GC Forms.

> What if there was an option to "Skip Review page" ?
I think for the sake of V1 and consistency with current forms, let's scope it so that _only_ **multi-page forms** get the Review page. And then we can simplify from there for V2 if we see it as a burden for users.

**Comment by Abi-Nada** [comment]
Testing out the PR environment:
Review page doesn't show for single pages ✅
But also can't seem to toggle to the next page for multiple pages? ❌

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeVVlQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d795d78829cb2a843f7b799dc67f0482389f19d8/Screen%20Recording%202024-07-30%20at%203.03.08%E2%80%AFPM.mov)

**Comment by thiessenp-cds** [comment]
Pushed a change that should fixe this

**Comment by Abi-Nada** [comment]
The fix works!!! Overall functionality looks good :)

But I no
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeU1qQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b46a667eea5782249754f9e1ba33ebc110564527/Screen%20Recording%202024-07-31%20at%202.57.38%E2%80%AFPM.mov)

ticed an Edge case while I was testing (but it could be less priority): when I deleted a second page and **immediately** go to Preview, I'm still able to click the "Continue" button and it gets wonky. (If I clicked on another page and then hit Preview, I don't run into this bug). **It's at the 1:00 minute mark of this video:**

But it's more likely related to #4135, so could potentially be outside the scope of this card?
